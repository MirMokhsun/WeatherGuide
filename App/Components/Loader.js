import React, { Component } from 'react'
import { Dimensions, Animated} from 'react-native'

const screenWidth = Math.round(Dimensions.get('window').width);
export default class Loader extends Component {
    state = {
        opacity: new Animated.Value(0),
    }

    onLoad = () => {
        Animated.timing(this.state.opacity, { toValue: 1, duration: 1000, }).start()
    }
    render() {
        return (
            <Animated.Image
                onLoad={this.onLoad}
                {...this.props}
                style={[
                    {
                        width: screenWidth,
                        height: screenWidth,
                        opacity: this.state.opacity,
                        transform: [
                            {
                                scale: this.state.opacity.interpolate({
                                    inputRange: [0, 1],
                                    outputRange: [0.25, 0.75]
                                })
                            }
                        ]
                    }
                ]}
            />
        )
    }
}