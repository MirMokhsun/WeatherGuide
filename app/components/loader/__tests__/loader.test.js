import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Loader from '../';

describe('Loader', () => {
    test('Loader snapshot', () => {
        const renderer = new ShallowRenderer();
        const snapshot = renderer.render(<Loader />);
        expect(snapshot).toMatchSnapshot();
    });
});