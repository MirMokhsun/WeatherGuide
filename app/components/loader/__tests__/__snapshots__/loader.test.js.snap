// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Loader snapshot 1`] = `
<AnimatedComponent
  source={
    Object {
      "testUri": "../../../app/assets/logos/cloudy-1.png",
    }
  }
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 250,
        "justifyContent": "center",
        "opacity": 0,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": 250,
      },
    ]
  }
/>
`;
